src/controllers/v1/members.ts

```diff
@@ -85,16 +118,24 @@ export class MembersController extends Controller {
     if (matchMemberByAlias) {
       throw new Conflict('member alias already exists')
     }
+    if (address === this.env.get('SELF_ADDRESS') && body.role) {
+      throw new BadRequest('cannot update role for self')
+    }
 
     if (!matchMemberByAddress) {
-      await this.db.insert('members', { address, alias })
+      await this.db.insert('members', { address, alias, role_id: role === undefined ? null : role.id })
       return {
         address,
         alias,
+        role: role === undefined ? undefined : role.role,
       }
     }
 
-    await this.db.update('members', { address }, { alias })
-    return { address, alias }
+    await this.db.update('members', { address }, { alias, role_id: role === undefined ? null : role.id })
```

Comment:
The logic here is wrong. If `role` is `undefined` it shouldn't be in the update at all. This will set the role to `null` in the db and overwrite it